### ISSUE TITLE
🐛 [BUG]: [postcss] Cannot read properties of undefined (reading 'config')

### ISSUE THREAD

**Issue by jesusvallez** [nxtensions/nxtensions#446]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What package does the issue affects?

@nxtensions/astro

### Describe the bug

Once you build the project with Tailwind, it doesn't seem to work and returns this error.

```
[postcss] Cannot read properties of undefined (reading 'config')
```

This problem occurs when we create a new repository from scratch.

### Expected behavior

I would expect the project to work smoothly

### Steps to reproduce

```
$ pnpx create-nx-workspace@latest nx-testing                                                                                                                                                         
.../Library/pnpm/store/v3/tmp/dlx-4578   |  +67 +++++++
.../Library/pnpm/store/v3/tmp/dlx-4578   | Progress: resolved 67, reused 67, downloaded 0, added 67, done

 >  NX   Let's create a new workspace [https://nx.dev/getting-started/intro]

✔ Which stack do you want to use? · react
✔ What framework would you like to use? · nextjs
✔ Integrated monorepo, or standalone project? · integrated
✔ Application name · nextjs-template
✔ Would you like to use the App Router (recommended)? · Yes
✔ Default stylesheet format · css
✔ Enable distributed caching to make your CI faster · No

 >  NX   Creating your v16.6.0 workspace.

   To make sure the command works reliably in all environments, and that the preset is applied correctly,
   Nx will run "pnpm install" several times. Please wait.

✔ Installing dependencies with pnpm
✔ Successfully created the workspace: nx-testing.

 >  NX   First time using Nx? Check out this interactive Nx tutorial.

   https://nx.dev/react-tutorial/1-code-generation
```

```
$ cd nx-testing
```

```
$ pnpm i -D -E @nxtensions/astro
Packages: +221
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Progress: resolved 1277, reused 1218, downloaded 0, added 221, done

devDependencies:
+ @nxtensions/astro 16.0.0

Done in 15.8s
```

(command used with Nx Console VSCode)
```
$ pnpm exec nx generate @nxtensions/astro:application --name=astro-template --addCypressTests=false --integrations=tailwind --no-interactive
>  NX  Generating @nxtensions/astro:application

UPDATE nx.json
UPDATE .gitignore
UPDATE .vscode/extensions.json
UPDATE package.json
CREATE tools/scripts/patch-nx-cli.js
UPDATE .npmrc
CREATE apps/astro-template/project.json
CREATE apps/astro-template/astro.config.mjs
CREATE apps/astro-template/public/favicon.svg
CREATE apps/astro-template/src/components/Card.astro
CREATE apps/astro-template/src/env.d.ts
CREATE apps/astro-template/src/layouts/Layout.astro
CREATE apps/astro-template/src/pages/index.astro
CREATE apps/astro-template/tsconfig.json
CREATE apps/astro-template/tailwind.config.cjs
✔ The modules directory at "/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules" will be removed and reinstalled from scratch. Proceed? (Y/n) · true
Recreating /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules
Packages: +1243
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Packages are hard linked from the content-addressable store to the virtual store.
  Content-addressable store is at: /Users/jesus/Library/pnpm/store/v3
  Virtual store is at:             node_modules/.pnpm
Progress: resolved 1302, reused 1243, downloaded 0, added 1243, done
node_modules/.pnpm/nx@16.6.0/node_modules/nx: Running postinstall script, done in 642ms

> @nx-testing/source@0.0.0 postinstall /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing
> node ./tools/scripts/patch-nx-cli.js

The Nx CLI was patched to allow importing ESM modules in the "@nxtensions/astro" project graph plugin.

dependencies:
+ @nx/next 16.6.0
+ next 13.4.1
+ react 18.2.0
+ react-dom 18.2.0
+ tslib 2.6.1

devDependencies:
+ @astrojs/tailwind 4.0.0
+ @nx/cypress 16.6.0
+ @nx/eslint-plugin 16.6.0
+ @nx/jest 16.6.0
+ @nx/js 16.6.0
+ @nx/linter 16.6.0
+ @nx/react 16.6.0
+ @nx/workspace 16.6.0
+ @nxtensions/astro 16.0.0
+ @testing-library/react 14.0.0
+ @types/jest 29.4.0
+ @types/node 18.14.2
+ @types/react 18.2.14
+ @types/react-dom 18.2.6
+ @typescript-eslint/eslint-plugin 5.60.1
+ @typescript-eslint/parser 5.60.1
+ astro 2.10.1
+ babel-jest 29.4.1
+ cypress 12.16.0
+ eslint 8.15.0
+ eslint-config-next 13.4.1
+ eslint-config-prettier 8.1.0
+ eslint-plugin-cypress 2.10.3
+ eslint-plugin-import 2.27.5
+ eslint-plugin-jsx-a11y 6.7.1
+ eslint-plugin-react 7.32.2
+ eslint-plugin-react-hooks 4.6.0
+ jest 29.4.1
+ jest-environment-jsdom 29.4.1
+ nx 16.6.0
+ prettier 2.6.2
+ tailwindcss 3.0.24 (3.3.3 is available)
+ ts-jest 29.1.0
+ ts-node 10.9.1
+ typescript 5.1.6

Done in 36.7s
```

> create and used a Tailwind component

```
$ nx dev astro-template
> nx run astro-template:dev

  🚀  astro  v2.10.1 started in 213ms

  ┃ Local    http://localhost:3000/
  ┃ Network  use --host to expose

[postcss] Cannot read properties of undefined (reading 'config')
22:51:51 [vite] Error when evaluating SSR module astro:scripts/page-ssr.js: failed to import "/@fs/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/@astrojs+tailwind@4.0.0_astro@2.10.1_tailwindcss@3.0.24_ts-node@10.9.1/node_modules/@astrojs/tailwind/base.css"
|- TypeError: [postcss] Cannot read properties of undefined (reading 'config')
    at getTailwindConfig (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/lib/setupTrackingContext.js:132:62)
    at /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/lib/setupTrackingContext.js:21:92
    at /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/processTailwindFeatures.js:38:11
    at plugins (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/index.js:33:58)
    at LazyResult.runOnRoot (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:357:16)
    at LazyResult.runAsync (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:286:26)
    at LazyResult.async (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:160:30)
    at LazyResult.then (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:436:17)

22:51:51 [vite] Error when evaluating SSR module /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/apps/astro-template/src/pages/index.astro: failed to import "astro:scripts/page-ssr.js"
|- TypeError: [postcss] Cannot read properties of undefined (reading 'config')
    at getTailwindConfig (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/lib/setupTrackingContext.js:132:62)
    at /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/lib/setupTrackingContext.js:21:92
    at /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/processTailwindFeatures.js:38:11
    at plugins (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/index.js:33:58)
    at LazyResult.runOnRoot (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:357:16)
    at LazyResult.runAsync (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:286:26)
    at LazyResult.async (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:160:30)
    at LazyResult.then (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:436:17)

 error   [postcss] Cannot read properties of undefined (reading 'config')
  File:
    /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/@astrojs+tailwind@4.0.0_astro@2.10.1_tailwindcss@3.0.24_ts-node@10.9.1/node_modules/@astrojs/tailwind/base.css
  Stacktrace:
TypeError: [postcss] Cannot read properties of undefined (reading 'config')
    at getTailwindConfig (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/lib/setupTrackingContext.js:132:62)
    at /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/lib/setupTrackingContext.js:21:92
    at /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/processTailwindFeatures.js:38:11
    at plugins (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/index.js:33:58)
    at LazyResult.runOnRoot (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:357:16)
    at LazyResult.runAsync (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:286:26)
    at LazyResult.async (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:160:30)
    at LazyResult.then (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:436:17)

[postcss] Cannot read properties of undefined (reading 'config')
 error   [postcss] Cannot read properties of undefined (reading 'config')
  File:
    /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/@astrojs+tailwind@4.0.0_astro@2.10.1_tailwindcss@3.0.24_ts-node@10.9.1/node_modules/@astrojs/tailwind/base.css
  Stacktrace:
TypeError: [postcss] Cannot read properties of undefined (reading 'config')
    at getTailwindConfig (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/lib/setupTrackingContext.js:132:62)
    at /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/lib/setupTrackingContext.js:21:92
    at /Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/processTailwindFeatures.js:38:11
    at plugins (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/tailwindcss@3.0.24_postcss@8.4.27_ts-node@10.9.1/node_modules/tailwindcss/lib/index.js:33:58)
    at LazyResult.runOnRoot (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:357:16)
    at LazyResult.runAsync (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:286:26)
    at LazyResult.async (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:160:30)
    at LazyResult.then (/Users/jesus/Documents/workspaceJS/Personal/New/nx-testing/node_modules/.pnpm/postcss@8.4.27/node_modules/postcss/lib/lazy-result.js:436:17)
 (x2)
```

### Environment

```
 >  NX   Report complete - copy this into the issue template

   Node   : 18.16.0
   OS     : darwin-arm64
   pnpm   : 8.6.11

   nx (global)        : 16.5.5
   nx                 : 16.6.0
   @nx/js             : 16.6.0
   @nx/jest           : 16.6.0
   @nx/linter         : 16.6.0
   @nx/workspace      : 16.6.0
   @nx/cypress        : 16.6.0
   @nx/devkit         : 16.6.0
   @nx/eslint-plugin  : 16.6.0
   @nx/next           : 16.6.0
   @nx/react          : 16.6.0
   @nrwl/tao          : 16.6.0
   @nx/web            : 16.6.0
   typescript         : 5.1.6
   ---------------------------------------
   Community plugins:
   @nxtensions/astro : 16.0.0
```

**Comment by firxworx** [comment]
I believe I may have a fix/workaround for you @jesusvallez 

See this comment I just added to my issue from earlier today:
https://github.com/nxtensions/nxtensions/issues/445#issuecomment-1664683087

**Comment by leosvelperez** [comment]
Thanks for reporting this!

The Astro `tailwind` integration recently released a breaking change in its v4.0.0 which causes this issue (https://github.com/withastro/astro/blob/main/packages/integrations/tailwind/CHANGELOG.md#400). Quoting it "This changes the Tailwind config loading behaviour where it is loaded from process.cwd() instead of the project root".

I'll push a fix later to handle that and generate the right configuration. Existing projects will need to add the path to the configuration.
